code/modules/projectiles/gun_attachables.dm

```diff
@@ -3274,14 +3278,31 @@ Defined in conflicts.dm of the #defines folder.
 	scatter_mod = SCATTER_AMOUNT_TIER_9
 	recoil_mod = RECOIL_AMOUNT_TIER_5
 
-/obj/item/attachable/bipod/Attach(obj/item/weapon/gun/G)
+/obj/item/attachable/bipod/Attach(obj/item/weapon/gun/gun)
 	..()
 
-	RegisterSignal(G, COMSIG_ITEM_DROPPED, PROC_REF(handle_drop))
+	if((GUN_FIREMODE_AUTOMATIC in gun.gun_firemode_list) || (gun.flags_gun_features & GUN_SUPPORT_PLATFORM))
+		var/given_action = FALSE
+		if(usr && (gun == usr.l_hand || gun == usr.r_hand))
```

Comment:
i know we're nullchecking this, but usr is always scary and bad in non-verbs. ideally, we pass an (optional) user var to `/attachable/proc/Attach` from `gun/proc/attach_to_gun` which we can drop in as a replacement for usr here